src/qibolab/_core/native.py

```diff
@@ -78,8 +80,11 @@ def R(self, theta: float = np.pi, phi: float = 0.0) -> PulseSequence:
 
         ``theta`` will be the angle of the rotation, while ``phi`` the angle that the rotation axis forms with x axis.
         """
-        assert self.RX is not None
-        return rotation(self.RX, theta, phi)
+        if self.RX90 != None:
+            return rotation(self.RX, 2 * theta, phi)
+        else:
+            assert self.RX is not None
+            return rotation(self.RX, theta, phi)
```

Comment:
```suggestion
        assert self.RX is not None
        return rotation(self.RX, theta, phi)
```
if you return in the former branch, you do not need the `else` :)